kasparovsyncd/sync/sync.go

```diff
@@ -405,38 +427,48 @@ func handleBlockAddedMsg(client *jsonrpc.Client, blockAdded *jsonrpc.BlockAddedM
 		return nil
 	}
 
-	err = fetchAndAddBlock(client, dbTx, blockHash)
+	addedBlockHashes, err := fetchAndAddBlock(client, dbTx, blockHash)
 	if err != nil {
 		return err
 	}
-	return dbTx.Commit()
+
+	err = dbTx.Commit()
+	if err != nil {
+		return err
+	}
+
+	for _, hash := range addedBlockHashes {
+		err := mqtt.PublishBlockAddedNotifications(hash)
```

Comment:
Why not call `mqtt.PublishBlockAddedNotifications` inside fetchAndAddBlock?
This way we'll be able to merge the call here and the call for missingBlockHashes.